webapp/src/ts/modules/reports/reports.component.ts

```diff
@@ -222,10 +232,10 @@ export class ReportsComponent implements OnInit, AfterViewInit, OnDestroy {
     return this.searchService
       .search('reports', this.filters, options)
       .then((reports) => this.addReadStatusService.updateReports(reports))
-      .then((updatedReports) => {
-        updatedReports = this.prepareReports(updatedReports);
-
-        this.reportsActions.updateReportsList(updatedReports);
+      .then(updatedReports => {
```

Comment:
Look how we're chaining promises in this function, and please think about the advantages of doing this, this is to run asynchronous code in a sequence (sync way).
You need to do the same with `prepareReports(updatedReports)` to keep data safe. I highly recommend to read and more importantly experiment with promises and asynchronous code because JavaScript in that regards won't warn you about undesired behaviours in code.

```
.then(reports => this.addReadStatusService.updateReports(reports))
.then(reports => this.prepareReports(reports))
.then(reports => {
  const updatedReports = this.reportsActions.updateReportsList(updatedReports);
  this.moreItems = updatedReports.length >= options.limit;
  ....
  ...
  ..
```